src/commands/synthetics/utils/internal.ts

```diff
@@ -253,11 +255,12 @@ export const validateAndParseOverrides = (overrides: string[] | undefined): Accu
           }
           break
 
-        // Convert to {[key: string]: string}
+        // Convert to StringObject
         case 'headers':
+        case 'variables':
           if (subKey) {
-            acc['headers'] = acc['headers'] ?? {}
-            acc['headers'][subKey] = value
+            acc[key] = acc[key] ?? {}
+            ;(acc[key] as StringObject)[subKey] = value
```

Comment:
Why is there a `;` at the start of the line here 🤔

Reply:
Otherwise you get this issue: `This expression is not callable. Type '{}' has no call signatures.` and it explicitly requires a semi-colon. The yarn format moves it to the start of the next line.